### PR TITLE
com_associations on install and on upgrade

### PR DESCRIPTION
**Pull request by jreys**
Pull Request for Issue #65  .
#### Summary of Changes

Adding com_associations to the installation and upgrade process for all 3 database systems.

**ATTENTION**: This needs to be properly tested on install and on update or joomla will break.

zip for tests 
https://github.com/joomla-projects/gsoc16_improved-multi-lingual/archive/a563f96818b3c2ac9286b11468be7e70d8f90526.zip

### CODE REVIEW

**Comment by infograf768** [comment]
Please do not take off the language files from the component language folder itself until this gets merged in core.

File: installation/sql/postgresql/joomla.sql [hunk 1]
```diff
@@ -501,6 +502,7 @@ INSERT INTO "#__extensions" ("extension_id", "name", "type", "element", "folder"
 (30, 'com_contenthistory', 'component', 'com_contenthistory', '', 1, 1, 1, 0, '', '', '', '', 0, '1970-01-01 00:00:00', 0, 0),
 (31, 'com_ajax', 'component', 'com_ajax', '', 1, 1, 1, 1, '', '', '', '', 0, '1970-01-01 00:00:00', 0, 0),
 (32, 'com_postinstall', 'component', 'com_postinstall', '', 1, 1, 1, 1, '', '', '', '', 0, '1970-01-01 00:00:00', 0, 0);
```
Comment (andrepereiradasilva) [inline installation/sql/postgresql/joomla.sql]: ok install on postgresql will be broken ... see the last character in this line.

File: installation/sql/mysql/joomla.sql [hunk 2]
```diff
@@ -514,6 +515,7 @@ INSERT INTO `#__extensions` (`extension_id`, `name`, `type`, `element`, `folder`
 (30, 'com_contenthistory', 'component', 'com_contenthistory', '', 1, 1, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
 (31, 'com_ajax', 'component', 'com_ajax', '', 1, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
 (32, 'com_postinstall', 'component', 'com_postinstall', '', 1, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
+(33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0);
```
Comment (andrepereiradasilva) [inline installation/sql/mysql/joomla.sql]: ok install on mysql will be broken ... see the last character in this line.

File: installation/sql/sqlazure/joomla.sql [hunk 3]
```diff
@@ -804,6 +806,8 @@ UNION ALL
 SELECT 31, 'com_ajax', 'component', 'com_ajax', '', 1, 1, 1, 1, '', '', '', '', 0, '1900-01-01 00:00:00', 0, 0
 UNION ALL
 SELECT 32, 'com_postinstall', 'component', 'com_postinstall', '', 1, 1, 1, 1, '', '', '', '', 0, '1900-01-01 00:00:00', 0, 0;
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/joomla.sql]: ok install on sqlazure will be broken ... see the last character in this line.

**Comment by andrepereiradasilva** [comment]
**ATTENTION**: This needs to be properly tested on install and on update or joomla will break.

File: administrator/components/com_admin/sql/updates/sqlazure/3.7.0-2016-09-01.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+SET IDENTITY_INSERT #__extensions  ON;
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/sqlazure/3.7.0-2016-09-01.sql]: `[#__extensions] ON;`
the same for all others

**Comment by andrepereiradasilva** [comment]
need to add the component to asset table in sample sql files too

File: installation/sql/sqlazure/sample_brochure.sql [hunk 5]
```diff
@@ -41,8 +41,10 @@ INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [titl
...
 INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES('41', '1', '79', '80', '1', 'com_joomlaupdate', 'com_joomlaupdate', '{"core.admin":[],"core.manage":[],"core.delete":[],"core.edit.state":[]}');
-INSERT INTO[#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES ('42', '1', '81', '82', '1', 'com_ajax', 'com_ajax', '{}');
-INSERT INTO[#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (43, 1, 83, 84, 1, 'com_postinstall', 'com_postinstall','{}');
+INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES ('42', '1', '81', '82', '1', 'com_ajax', 'com_ajax', '{}');
+INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (43, 1, 83, 84, 1, 'com_postinstall', 'com_postinstall','{}');
+INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (48, 1, 85, 86, 1, 'com_associations', 'com_associations', '{}');
+
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/sample_brochure.sql]: please remove this extra line you added.

File: installation/sql/sqlazure/sample_data.sql [hunk 6]
```diff
@@ -43,6 +43,8 @@ INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (37, 1, 71, 72, 1, 'com_contenthistory', 'com_contenthistory', '{}');
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (38, 1, 73, 74, 1, 'com_ajax', 'com_ajax', '{}');
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (39, 1, 75, 76, 1, 'com_postinstall', 'com_postinstall', '{}');
+INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (40, 1, 77, 78, 1, 'com_associations', 'com_associations', '{}');
+
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/sample_data.sql]: please remove this extra line you added.

File: installation/sql/sqlazure/sample_learn.sql [hunk 7]
```diff
@@ -170,6 +170,8 @@ INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (176, 1, 424, 425, 1, 'com_contenthistory', 'com_contenthistory', '{}');
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (177, 1, 426, 427, 1, 'com_ajax', 'com_ajax', '{}');
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (178, 1, 428, 429, 1, 'com_postinstall', 'com_postinstall', '{}');
+INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (179, 1, 430, 431, 1, 'com_associations', 'com_associations', '{}');
+
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/sample_learn.sql]: please remove this extra line you added.

File: installation/sql/sqlazure/sample_testing.sql [hunk 8]
```diff
@@ -171,6 +171,8 @@ INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (175, 1, 426, 427, 1, 'com_contenthistory', 'com_contenthistory', '{}');
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (176, 1, 428, 429, 1, 'com_ajax', 'com_ajax', '{}');
 INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (177, 1, 429, 430, 1, 'com_postinstall', 'com_postinstall', '{}');
+INSERT [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (178, 1, 431, 432, 1, 'com_associations', 'com_associations', '{}');
+
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/sample_testing.sql]: please remove this extra line you added.

File: installation/sql/mysql/sample_data.sql [hunk 9]
```diff
@@ -73,7 +73,8 @@ INSERT IGNORE INTO `#__assets` (`id`, `parent_id`, `lft`, `rgt`, `level`, `name`
 (59, 18, 82, 83, 2, 'com_modules.module.92', 'Image Module', '{"core.delete":{"6":1},"core.edit":{"6":1,"4":1},"core.edit.state":{"6":1,"5":1}}'),
 (60, 18, 84, 85, 2, 'com_modules.module.93', 'Search', '{"core.delete":{"6":1},"core.edit":{"6":1,"4":1},"core.edit.state":{"6":1,"5":1}}'),
 (61, 27, 19, 20, 3, 'com_content.article.1', 'Getting Started', '{"core.delete":{"6":1},"core.edit":{"6":1,"4":1},"core.edit.state":{"6":1,"5":1}}'),
-(62, 1, 117, 118, 1, '#__ucm_content.1', '#__ucm_content.1', '[]');
+(62, 1, 117, 118, 1, '#__ucm_content.1', '#__ucm_content.1', '[]'),
+(63, 1, 119, 120, 1, 'com_associations', 'com_associations', '{}'),
```
Comment (andrepereiradasilva) [inline installation/sql/mysql/sample_data.sql]: look at the last char...

File: administrator/components/com_admin/sql/updates/mysql/3.7.0-2016-09-01.sql [hunk 10]
```diff
@@ -1,2 +1,4 @@
...
-(33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0);
\ No newline at end of file
+INSERT INTO "#__extensions" ("extension_id", "name", "type", "element", "folder", "client_id", "enabled", "access", "protected", "manifest_cache", "params", "custom_data", "system_data", "checked_out", "checked_out_time", "ordering", "state") VALUES
+(33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '1970-01-01 00:00:00', 0, 0);
+INSERT INTO "#__menu" ("menutype", "title", "alias", "note", "path", "link", "type", "published", "parent_id", "level", "component_id", "checked_out", "checked_out_time", "browserNav", "access", "img", "template_style_id", "params", "lft", "rgt", "home", "language", "client_id") VALUES
+('menu', 'com_associations', 'Multilingual Associations', '', 'Multilingual Associations', 'index.php?option=com_associations', 'component', 1, 1, 1, 33, 0, '1970-01-01 00:00:00', 0, 1, 'class:associations', 0, '', 43, 44, 0, '*', 1),
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/mysql/3.7.0-2016-09-01.sql]: look a t the last character

File: administrator/components/com_admin/sql/updates/postgresql/3.7.0-2016-09-01.sql [hunk 11]
```diff
@@ -1,2 +1,4 @@
 INSERT INTO "#__extensions" ("extension_id", "name", "type", "element", "folder", "client_id", "enabled", "access", "protected", "manifest_cache", "params", "custom_data", "system_data", "checked_out", "checked_out_time", "ordering", "state") VALUES
 (33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '1970-01-01 00:00:00', 0, 0);
+INSERT INTO "#__menu" ("menutype", "title", "alias", "note", "path", "link", "type", "published", "parent_id", "level", "component_id", "checked_out", "checked_out_time", "browserNav", "access", "img", "template_style_id", "params", "lft", "rgt", "home", "language", "client_id") VALUES
+('menu', 'com_associations', 'Multilingual Associations', '', 'Multilingual Associations', 'index.php?option=com_associations', 'component', 1, 1, 1, 33, 0, '1970-01-01 00:00:00', 0, 1, 'class:associations', 0, '', 43, 44, 0, '*', 1),
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/postgresql/3.7.0-2016-09-01.sql]: look at the last character

File: administrator/components/com_admin/sql/updates/sqlazure/3.7.0-2016-09-01.sql [hunk 12]
```diff
@@ -3,4 +3,7 @@ SET IDENTITY_INSERT [#__extensions] ON;
 INSERT INTO [#__extensions] ([extension_id], [name], [type], [element], [folder], [client_id], [enabled], [access], [protected], [manifest_cache], [params], [custom_data], [system_data], [checked_out], [checked_out_time], [ordering], [state])
 SELECT 33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '1900-01-01 00:00:00', 0, 0;
 
-SET IDENTITY_INSERT [#__extensions] OFF;
\ No newline at end of file
+SET IDENTITY_INSERT [#__extensions] OFF;
+
+INSERT INTO [#__menu] ([menutype], [title], [alias], [note], [path], [link], [type], [published], [parent_id], [level], [component_id], [checked_out], [checked_out_time], [browserNav], [access], [img], [template_style_id], [params], [lft], [rgt], [home], [language], [client_id])
+SELECT 'menu', 'com_associations', 'Multilingual Associations', '', 'Multilingual Associations', 'index.php?option=com_associations', 'component', 1, 1, 1, 33, 0, '1900-01-01 00:00:00', 0, 1, 'class:associations', 0, '', 43, 44, 0, '*', 1
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/sqlazure/3.7.0-2016-09-01.sql]: look at the last character (NEEDS ;)

File: administrator/components/com_admin/sql/updates/mysql/3.7.0-2016-09-01.sql (hunk 10 above)
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/mysql/3.7.0-2016-09-01.sql]: mysql quotes are not `"`!

File: installation/sql/mysql/sample_blog.sql [hunk 13]
```diff
@@ -56,7 +56,8 @@ INSERT IGNORE INTO `#__assets` (`id`, `parent_id`, `lft`, `rgt`, `level`, `name`
 (44, 1, 79, 80, 1, 'com_tags', 'com_tags', '{"core.admin":{"8":1},"core.manage":{"7":1},"core.create":{"6":1,"3":1},"core.delete":{"6":1},"core.edit":{"6":1,"4":1},"core.edit.state":{"6":1,"5":1}}'),
 (45, 1, 81, 82, 1, 'com_contenthistory', 'com_contenthistory', '{}'),
 (46, 1, 83, 84, 1, 'com_ajax', 'com_ajax', '{}'),
-(47, 1, 85, 86, 1, 'com_postinstall', 'com_postinstall', '{}');
+(47, 1, 85, 86, 1, 'com_postinstall', 'com_postinstall', '{}'),
+(49, 1, 87, 88, 1, 'com_associations', 'com_associations', '{}');
```
Comment (andrepereiradasilva) [inline installation/sql/mysql/sample_blog.sql]: 48?

File: installation/sql/postgresql/joomla.sql [hunk 14]
```diff
@@ -1225,12 +1227,13 @@ INSERT INTO "#__menu" ("id", "menutype", "title", "alias", "note", "path", "link
 (13, 'menu', 'com_newsfeeds', 'News Feeds', '', 'News Feeds', 'index.php?option=com_newsfeeds', 'component', 0, 1, 1, 17, 0, '1970-01-01 00:00:00', 0, 0, 'class:newsfeeds', 0, '', 23, 28, 0, '*', 1),
 (14, 'menu', 'com_newsfeeds_feeds', 'Feeds', '', 'News Feeds/Feeds', 'index.php?option=com_newsfeeds', 'component', 0, 13, 2, 17, 0, '1970-01-01 00:00:00', 0, 0, 'class:newsfeeds', 0, '', 24, 25, 0, '*', 1),
 (15, 'menu', 'com_newsfeeds_categories', 'Categories', '', 'News Feeds/Categories', 'index.php?option=com_categories&extension=com_newsfeeds', 'component', 0, 13, 2, 6, 0, '1970-01-01 00:00:00', 0, 0, 'class:newsfeeds-cat', 0, '', 26, 27, 0, '*', 1),
-(16, 'menu', 'com_redirect', 'Redirect', '', 'Redirect', 'index.php?option=com_redirect', 'component', 0, 1, 1, 24, 0, '1970-01-01 00:00:00', 0, 0, 'class:redirect', 0, '', 29, 30, 0, '*', 1),
+(16, 'menu', 'com_redirect', 'Redirect', '', 'Redirect', 'index.php?option=com_redirect', 'component', 0, 1, 1, 24, 0, '1970424-01-01 00:00:00', 0, 0, 'class:redirect', 0, '', 29, 30, 0, '*', 1),
```
Comment (andrepereiradasilva) [inline installation/sql/postgresql/joomla.sql]: 1970424-01-01 00:00:00 ????

File: installation/sql/sqlazure/sample_brochure.sql [hunk 15]
```diff
@@ -86,35 +87,36 @@ SET IDENTITY_INSERT [#__content] OFF;
...
+INSERT INTO [#__menu] ( [id], [menutype], [title], [alias], [note], [path], [link], [type], [published], [parent_id], [level], [component_id], [checked_out], [checked_out_time], [browserNav], [access], [img], [template_style_id], [params], [lft], [rgt], [home], [language], [client_id]) VALUES(109, 'mainmenu', 'Create an Article', 'create-an-article', '', 'create-an-article', 'index.php?option=com_content&view=form&layout=edit', 'component', 1, 1, 1, 22, 0, '1900-01-01T00:00:00.000', 0, 3, '', 0, '{"enable_category":"0","catid":"2","menu-anchor_title":"","menu-anchor_css":"","menu_image":"","menu_text":1,"page_title":"","show_page_heading":0,"page_heading":"","pageclass_sfx":"","menu-meta_description":"","menu-meta_keywords":"","robots":"","secure":0}', 13, 14, 0, '*', 0);
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/sample_brochure.sql:119]: please don't do other changes in this PR is more hard to review if you do.

File: installation/sql/sqlazure/sample_brochure.sql [hunk 16]
```diff
@@ -41,8 +41,9 @@ INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [titl
...
 INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES('41', '1', '79', '80', '1', 'com_joomlaupdate', 'com_joomlaupdate', '{"core.admin":[],"core.manage":[],"core.delete":[],"core.edit.state":[]}');
-INSERT INTO[#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES ('42', '1', '81', '82', '1', 'com_ajax', 'com_ajax', '{}');
-INSERT INTO[#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (43, 1, 83, 84, 1, 'com_postinstall', 'com_postinstall','{}');
+INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES ('42', '1', '81', '82', '1', 'com_ajax', 'com_ajax', '{}');
+INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (43, 1, 83, 84, 1, 'com_postinstall', 'com_postinstall','{}');
+INSERT INTO [#__assets] ([id], [parent_id], [lft], [rgt], [level], [name], [title], [rules]) VALUES (48, 1, 85, 86, 1, 'com_associations', 'com_associations', '{}');
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/sample_brochure.sql]: please don't do other changes in this PR is more hard to review if you do.

File: administrator/components/com_admin/sql/updates/mysql/3.7.0-2016-09-01.sql [hunk 17]
```diff
@@ -0,0 +1,4 @@
...
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/mysql/3.7.0-2016-09-01.sql]: look at last character

File: administrator/components/com_admin/sql/updates/postgresql/3.7.0-2016-09-01.sql [hunk 18]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO "#__extensions" ("extension_id", "name", "type", "element", "folder", "client_id", "enabled", "access", "protected", "manifest_cache", "params", "custom_data", "system_data", "checked_out", "checked_out_time", "ordering", "state") VALUES
+(33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '1970-01-01 00:00:00', 0, 0);
+INSERT INTO "#__menu" ("menutype", "title", "alias", "note", "path", "link", "type", "published", "parent_id", "level", "component_id", "checked_out", "checked_out_time", "browserNav", "access", "img", "template_style_id", "params", "lft", "rgt", "home", "language", "client_id") VALUES
+('menu', 'com_associations', 'Multilingual Associations', '', 'Multilingual Associations', 'index.php?option=com_associations', 'component', 1, 1, 1, 33, 0, '1970-01-01 00:00:00', 0, 1, 'class:associations', 0, '', 43, 44, 0, '*', 1),
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/postgresql/3.7.0-2016-09-01.sql]: look at last character

File: administrator/components/com_admin/sql/updates/sqlazure/3.7.0-2016-09-01.sql [hunk 19]
```diff
@@ -0,0 +1,9 @@
+SET IDENTITY_INSERT [#__extensions] ON;
+
+INSERT INTO [#__extensions] ([extension_id], [name], [type], [element], [folder], [client_id], [enabled], [access], [protected], [manifest_cache], [params], [custom_data], [system_data], [checked_out], [checked_out_time], [ordering], [state])
+SELECT 33, 'com_associations', 'component', 'com_associations', '', 1, 1, 1, 1, '', '', '', '', 0, '1900-01-01 00:00:00', 0, 0;
+
+SET IDENTITY_INSERT [#__extensions] OFF;
+
+INSERT INTO [#__menu] ([menutype], [title], [alias], [note], [path], [link], [type], [published], [parent_id], [level], [component_id], [checked_out], [checked_out_time], [browserNav], [access], [img], [template_style_id], [params], [lft], [rgt], [home], [language], [client_id])
+SELECT 'menu', 'com_associations', 'Multilingual Associations', '', 'Multilingual Associations', 'index.php?option=com_associations', 'component', 1, 1, 1, 33, 0, '1900-01-01 00:00:00', 0, 1, 'class:associations', 0, '', 43, 44, 0, '*', 1
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/sql/updates/sqlazure/3.7.0-2016-09-01.sql]: look at last character
also you need to start/end SET IDENTITY_INSERT [#__menu] ON/OFF;

**Comment by alikon** [comment]
for the recent merge your .zip is obsolete

**Comment by infograf768** [comment]
New Joomla install OK on mysqli, no sample data

**Comment by infograf768** [comment]
If I install using test sample data and then multilang during install I get an error in the side by side (did not change anything else)

screen shot 2016-08-10 at 10 39 23

**Comment by infograf768** [comment]
Also
Administrator permissions are different in com_associations from other components:

Exemple for contacts
screen shot 2016-08-10 at 10 44 40

For associations
screen shot 2016-08-10 at 10 46 08

Therefore missing Allowed for `Configure ACL & Options`

**Comment by infograf768** [comment]
Testing by installing with normal sample data (sample_data.sql), I do not get the error in the side by side.

**Comment by alikon** [comment]
on postgresql with a standard multilanguage install went fine (i.e no sample/test/blog data)

**Comment by infograf768** [comment]
Concerning default permission for administrator, I tried to change in joomla.sql in the assets table to
`(55, 1, 103, 104, 1, 'com_associations', 'com_associations', '{"core.admin":{"7":1},"core.manage":{"6":1}');`

But I did not get the expected result.

**Comment by andrepereiradasilva** [comment]
@jreys  you need to fix conflicts because the 'n' PR ( https://github.com/joomla/joomla-cms/commit/540a3bc1f88aa33e00df7f17f54be2218314b819 ) was merged into the core

**Comment by alikon** [comment]
@infograf768  for https://github.com/joomla-projects/gsoc16_improved-multi-lingual/pull/77#issuecomment-238893784

the right update should be  

``` sql
update #__assets
set rules ='{"core.admin":{"7":1},"core.manage":{"6":1}}'
where id = your_associaction_id
```

 and in this way i got 

assoc

`

**Comment by infograf768** [comment]
@alikon 
For udaptes most ceratinly
But for new installs?
What I tested above was for a new install

**Comment by alikon** [comment]
as discussed on chat we need a fix on .sql files

**Comment by andrepereiradasilva** [comment]
that query will not work
you don't have the asset id. you have to use the WHERE "name" = 'com_associations'

``` sql
UPDATE `#__assets`
SET `rules` ='{"core.admin":{"7":1},"core.manage":{"6":1}}'
WHERE `name` = 'com_associations'
```

**Comment by andrepereiradasilva** [comment]
But also you can't use that in an update because the group ids can be others.
Group 7 is Administrator and 6 is Manager but only on install.

Sometime ago i made a PoC for similar job on update
See https://github.com/infograf768/joomla-cms/pull/30/files#diff-e3c745afdbb43914c4f183a8c1587346R1619

**Comment by alikon** [comment]
@andrepereiradasilva  the update query was just an example  to better understand the fix needed

**Comment by andrepereiradasilva** [comment]
ok

**Comment by andrepereiradasilva** [comment]
@jreys  fix conflicts please
we also need to find a way to solve the ACL on update before this can be merged

**Comment by andrepereiradasilva** [comment]
merged as decided

**Pull request merged**